.github/workflows/publish-engine.yml

```diff
@@ -0,0 +1,28 @@
+name: publish-engine
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+      - "!sdk/go/**"
```

Comment:
This should not be needed -- I hope?

Reply:
My thinking was that we don't want to rebuild a new engine image if all we did was tag a new sdk release. I.e. don't want to build an engine image like `ghcr.io/dagger/engine:sdk/go/v0.4.1` (I don't think that's even a valid image tag?).

Instead this workflow should only run when we do a new engine tag or on main.

Right? Easily could be missing something...

Reply:
Oh also, just to clarify, since we hardcode the image SHAs in the SDKs, the `tag` is not functionally important to the provisioning.

The only reason to tag with the engine version is that it makes it easier for us to track which engine image is associated with which version of the engine. Just reduces risk of human error when updating SDKs.

Reply:
> Instead this workflow should only run when we do a new engine tag or on main.

I **think** that is captured by `v*` -- `sdk/go/vXXX` shouldn't match the `v*` pattern

Reply:
omg duh..... okay, that makes perfect sense. Yeah I also hope the rest aren't needed 😂 I will remove them.